hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1426,15 +1428,15 @@ def new_shrinker(
 
     def passing_choice_sequences(
         self, prefix: Sequence[IRNode] = ()
-    ) -> frozenset[bytes]:
+    ) -> frozenset[tuple[IRNode, ...]]:
         """Return a collection of choice sequence nodes which cause the test to pass.
         Optionally restrict this by a certain prefix, which is useful for explain mode.
         """
         return frozenset(
-            result.ir_nodes
+            cast(ConjectureResult, result).ir_nodes
             for key in self.__data_cache
             if (result := self.__data_cache[key]).status is Status.VALID
-            and startswith(result.ir_nodes, prefix)
+            and startswith(cast(ConjectureResult, result).ir_nodes, prefix)
```

Comment:
I think we can make this better with a `is_valid(data: ConjectureResult | Overrun) -> TypeGuard[ConjectureResult]` in the future